tests/components/ialarm/test_init.py

```diff
@@ -0,0 +1,85 @@
+"""Test the Antifurto365 iAlarm init."""
+from unittest.mock import Mock, patch
+from uuid import uuid4
+
+import pytest
+
+from homeassistant.components.ialarm.const import DOMAIN
+from homeassistant.config_entries import (
+    ENTRY_STATE_LOADED,
+    ENTRY_STATE_NOT_LOADED,
+    ENTRY_STATE_SETUP_RETRY,
+)
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(name="ialarm_api")
+def ialarm_api_fixture():
+    """Set up IAlarm API fixture."""
+    with patch("homeassistant.components.ialarm.IAlarm") as mock_ialarm_api:
+        yield mock_ialarm_api
+
+
+@pytest.fixture(name="mock_config_entry")
+def mock_config_fixture():
+    """Return a fake config entry."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        data={CONF_HOST: "192.168.10.20", CONF_PORT: 18034},
+        entry_id=str(uuid4()),
+    )
+
+
+async def test_setup_entry(hass, ialarm_api, mock_config_entry):
+    """Test setup entry."""
+    ialarm_api.return_value.get_mac = Mock(return_value="00:00:54:12:34:56")
+
+    mock_config_entry.add_to_hass(hass)
+    await async_setup_component(
+        hass,
+        DOMAIN,
+        {
```

Comment:
We don't support config yaml for this integration. Passing the config here doesn't do anything. Set up the integration with `hass.config_entries.async_setup` instead.